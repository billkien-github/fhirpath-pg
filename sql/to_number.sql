
SELECT fhirpath_as_number('{"a":{"b": {"c": 5.3}}}', '.a.b.c', 'decimal', 'min') + 0.1; 
SELECT fhirpath_as_number('{"a":{"b": {"c": 5}}}', '.a.b.c', 'integer', 'min') + 1;
SELECT fhirpath_as_number('{"a":{"b": {"c": 5}}}', '.a.b.c', 'positiveInt', 'min') + 1;
SELECT fhirpath_as_number('{"a":{"b": {"c": 5}}}', '.a.b.c', 'unsignedInt', 'min') + 1;
SELECT fhirpath_as_number('{"a":{"b": {"c": "ups"}}}', '.a.b.c', 'decimal', 'min');
SELECT fhirpath_as_number('{"a":{"b": {"value": 5.1}}}', '.a.b', 'Age', 'min') + 0.05;
SELECT fhirpath_as_number('{"a":{"b": {"value": 5.1}}}', '.a.b', 'Count', 'min') + 0.05;
SELECT fhirpath_as_number('{"a":{"b": {"value": 5.1}}}', '.a.b', 'Money', 'min') + 0.05;
SELECT fhirpath_as_number('{"a":{"b": {"value": 5.1}}}', '.a.b', 'Distance', 'min') + 0.05;
SELECT fhirpath_as_number('{"a":{"b": {"value": 5.1}}}', '.a.b', 'SimpleQuantity', 'min') + 0.05;
SELECT fhirpath_as_number('{"a":{"b": {"value": 5.1}}}', '.a.b', 'Duration', 'min') + 0.05;
SELECT fhirpath_as_number('{"a":{"b": {"value": 5.1}}}', '.a.b', 'Quantity', 'min') + 0.05;
SELECT fhirpath_as_number('{"Quantity":{"b": {"value": 5.1}}}', '.Quantity.b', 'Quantity', 'min') + 0.05;
SELECT fhirpath_as_number('{"a":{"b": {"value": "ups"}}}', '.a.b', 'Quantity', 'min') + 0.05;

SELECT fhirpath_as_number('{"a":{"b": {"value": [5.1, 6.1]}}}', '.a.b.value', 'decimal', 'min') + 0.05;
SELECT fhirpath_as_number('{"a":{"b": {"value": [5.1, 6.1]}}}', '.a.b.value', 'decimal', 'max') + 0.05;

SELECT fhirpath_as_number('{"a":{"b": [{"value": 5.1},{"value": 6.1}]}}', '.a.b', 'Quantity', 'min') + 0.05;
SELECT fhirpath_as_number('{"a":{"b": [{"value": 5.1},{"value": 6.1}]}}', '.a.b', 'Quantity', 'max') + 0.05;
SELECT fhirpath_as_number('{"a":{"b": [{"value": 7.1},{"value": 6.1}]}}', '.a.b', 'Quantity', 'max') + 0.05;

SELECT fhirpath_as_number('{"a":[7.1, 6.1, 100, 1.1]}', '.a', 'decimal', 'max') + 0.05;
SELECT fhirpath_as_number('{"a":[7.1, 6.1, 100, 1.1]}', '.a', 'decimal', 'min') + 0.05;

