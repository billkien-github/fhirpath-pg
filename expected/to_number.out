SELECT fhirpath_as_number('{"a":{"b": {"c": 5.3}}}', '.a.b.c', 'decimal') + 0.1; 
 ?column? 
----------
      5.4
(1 row)

SELECT fhirpath_as_number('{"a":{"b": {"c": 5}}}', '.a.b.c', 'integer') + 1;
 ?column? 
----------
        6
(1 row)

SELECT fhirpath_as_number('{"a":{"b": {"c": 5}}}', '.a.b.c', 'positiveInt') + 1;
 ?column? 
----------
        6
(1 row)

SELECT fhirpath_as_number('{"a":{"b": {"c": 5}}}', '.a.b.c', 'unsignedInt') + 1;
 ?column? 
----------
        6
(1 row)

SELECT fhirpath_as_number('{"a":{"b": {"c": "ups"}}}', '.a.b.c', 'decimal');
 fhirpath_as_number 
--------------------
                   
(1 row)

SELECT fhirpath_as_number('{"a":{"b": {"value": 5.1}}}', '.a.b', 'Age') + 0.05;
 ?column? 
----------
     5.15
(1 row)

SELECT fhirpath_as_number('{"a":{"b": {"value": 5.1}}}', '.a.b', 'Count') + 0.05;
 ?column? 
----------
     5.15
(1 row)

SELECT fhirpath_as_number('{"a":{"b": {"value": 5.1}}}', '.a.b', 'Money') + 0.05;
 ?column? 
----------
     5.15
(1 row)

SELECT fhirpath_as_number('{"a":{"b": {"value": 5.1}}}', '.a.b', 'Distance') + 0.05;
 ?column? 
----------
     5.15
(1 row)

SELECT fhirpath_as_number('{"a":{"b": {"value": 5.1}}}', '.a.b', 'SimpleQuantity') + 0.05;
 ?column? 
----------
     5.15
(1 row)

SELECT fhirpath_as_number('{"a":{"b": {"value": 5.1}}}', '.a.b', 'Duration') + 0.05;
 ?column? 
----------
     5.15
(1 row)

SELECT fhirpath_as_number('{"a":{"b": {"value": 5.1}}}', '.a.b', 'Quantity') + 0.05;
 ?column? 
----------
     5.15
(1 row)

SELECT fhirpath_as_number('{"a":{"b": {"value": "ups"}}}', '.a.b', 'Quantity') + 0.05;
 ?column? 
----------
         
(1 row)

